### PR TITLE
document DynamicConfigProvider for kafka consumer properties

### PR DESCRIPTION
**Pull request by himanshug**
documentation for #10309

### CODE REVIEW

File: docs/development/extensions-core/kafka-ingestion.md [hunk 1]
```diff
@@ -134,7 +134,7 @@ A sample supervisor spec is shown below:
 |-----|----|-----------|--------|
 |`topic`|String|The Kafka topic to read from. This must be a specific topic as topic patterns are not supported.|yes|
 |`inputFormat`|Object|[`inputFormat`](../../ingestion/data-formats.md#input-format) to specify how to parse input data. See [the below section](#specifying-data-format) for details about specifying the input format.|yes|
-|`consumerProperties`|Map<String, Object>|A map of properties to be passed to the Kafka consumer. This must contain a property `bootstrap.servers` with a list of Kafka brokers in the form: `<BROKER_1>:<PORT_1>,<BROKER_2>:<PORT_2>,...`. Users can set `isolation.level` `read_uncommitted` here if don't need Druid to consume transactional topics or need Druid to consume older versions of Kafka. For SSL connections, the `keystore`, `truststore` and `key` passwords can be provided as a [Password Provider](../../operations/password-provider.md) or String password.|yes|
+|`consumerProperties`|Map<String, Object>|A map of properties to be passed to the Kafka consumer. See next section for more information.|yes|
```
Comment (jihoonson) [inline docs/development/extensions-core/kafka-ingestion.md:137]:
```suggestion
|`consumerProperties`|Map<String, Object>|A map of properties to be passed to the Kafka consumer. See [next section](#more-on-consumerproperties) for more information.|yes|
```

File: docs/development/extensions-core/kafka-ingestion.md [hunk 2]
```diff
@@ -147,6 +147,18 @@ A sample supervisor spec is shown below:
...
+
+This must contain a property `bootstrap.servers` with a list of Kafka brokers in the form: `<BROKER_1>:<PORT_1>,<BROKER_2>:<PORT_2>,...`.
+By default, `isolation.level` is set to `read_committed`. It should be set to `read_uncommitted` if you don't want Druid to consume only committed transactions or working with older versions of Kafka servers with no Transactions support.
+
+There are few cases that require fetching few/all of consumer properties at runtime e.g. when `bootstrap.servers` is not known upfront or not static, to enable SSL connections users might have to provide passwords for `keystore`, `truststore` and `key` secretly.
+For such consumer properties, user can implement a [DynamicConfigProvider](../../operations/dynamic-config-provider.md) to supply them at runtime, by adding
+`druid.dynamic.config.provider`=`{"type": "<registered_dynamic_config_provider_name>", ...}`
+in consumerProperties map.
+
+Note: In older Druid versions, for SSL connections, the `keystore`, `truststore` and `key` passwords can also be provided as a [Password Provider](../../operations/password-provider.md). This is deprecated.
```
Comment (jihoonson) [inline docs/development/extensions-core/kafka-ingestion.md:160]: It would be better to mention Druid versions. Maybe `In 0.20.0 or older Druid versions`?

Comment (himanshug) [reply docs/development/extensions-core/kafka-ingestion.md:160]:
```suggestion
Note: In 0.20.0 or older Druid versions, for SSL connections, the `keystore`, `truststore` and `key` passwords can also be provided as a [Password Provider](../../operations/password-provider.md). This is deprecated.
```

**Comment by himanshug** [comment]
thanks @jihoonson for quick review, applied both changes.

**Review by jihoonson — APPROVED** [review]
@himanshug thanks for quick fix. +1 after CI.

**Pull request merged**